### DISCUSSION TITLE
Problem when using pnpm

### DISCUSSION THREAD

**Discussion by ZeroGDrive** [cyrilletuzi/vscode-angular-schematics#911] [Q&A] [answered]
Hi,

# Checks before posting an issue

- [x] I have read _all_ the documentation included in the extension itself as a walkthrough, including the [troubleshooting guide](https://github.com/cyrilletuzi/vscode-angular-schematics/blob/main/walkthroughs/troubleshooting.md) (from the VS Code menu: View > Command Palette >search "Documentation" > choose "Angular Schematics: Documentation and tutorial")
- [x] I have checked in already:
  - [opened issues](https://github.com/cyrilletuzi/vscode-angular-schematics/issues)
  - [closed issues](https://github.com/cyrilletuzi/vscode-angular-schematics/issues?q=is%3Aissue+is%3Aclosed)
  - [questions answered](https://github.com/cyrilletuzi/vscode-angular-schematics/discussions/categories/q-a) in the new "Discussions" tab
- [x] My project meets the requirements indicated in README
- [x] My workspace is in [trusted mode](https://code.visualstudio.com/docs/editor/workspace-trust), not restricted mode
- [x] I understand this extension is just *using* Angular CLI schematics, it does *not* create files by itself, so any issue with the schematics should be reported in the relevant repository:
  - [Angular CLI issue tracker](https://github.com/angular/angular-cli/issues)
- [x] I launched `ng g component hello` in the *VS Code* Terminal inside my project, and it works
- [x] I looked at the extension's logs ("Angular Schematics" Output channel, second tab left to the Terminal), and tried to fix my project's configuration accordingly
- [x] I understand I am writing to a *human being*, doing this project *unpaid*
- [x] I will be involved in my own issue, by answering questions and informing if the issue is resolved

# Configuration

- VS Code version (only the last version is supported): 1.82
- Extension version: 5.4.2
- OS: mac os 13.5.2
- Are you using VS Code in normal mode or in a special mode (web version, WSL, SSH/remote, etc.): normal mode
- Angular version (`ng version`, only LTS versions are supported): 16.1

# Description of the issue
When using pnpm in the project the extension doesn't work at all.

# Extension logs

"@schematics/angular" should be present in a correctly installed Angular project.
(Note: I even manually installed @schematics/angular but it doesn't work)

# How to reproduce the issue
Just use an angular project with pnpm and you will notice the issue. the issue is present in all projects I've been working on that uses pnpm. and when reverting to npm or yarn the extension works perfectly fine.

**Comment by cyrilletuzi** [comment]
Hi,

This is not an issue, it is working as intended. The extension needs to read the schematics information from `node_modules/@schematics/angular`. It is too complicated and time-consuming to look in all the fantasy, changing-at-every-version and depending-on-x-y-and-x places where alternative package managers put their dependencies, especially when they do not provide an API to do so.

For information, Schematics Pro, an extended version of the extension, uses its own schematics, and thus does not need to read `node_modules` and thus is not impacted by which package manager the project uses (so it works with pnpm). You will find all the details on the [dedicated website](https://www.cyrilletuzi.com/schematics-pro/).

**Comment by cyrilletuzi** [comment, marked as answer]
Hi @ZeroGDrive, just to let you know that the latest version (v6) of the extension now works with any package manager.
